src/Geno/C++/GUI/Modals/OpenFileModal.cpp

```diff
@@ -97,10 +99,38 @@ void OpenFileModal::OnClose( void )
 
 } // OnClose
 
+void OpenFileModal::UpdateSplitStingBuffer()
+{
+	m_SplitStringBuffer.clear();
+	m_SplitStringBuffer = StringAux::SplitString( m_CurrentDirectory.string(), "/\\" );
+
+} // UpdateSplitStingBuffer
+
 //////////////////////////////////////////////////////////////////////////
 
 void OpenFileModal::UpdateDerived( void )
 {
+	UpdateSplitStingBuffer();
+
+	for ( const std::string& buttonName : m_TempSplitStringBuffer )
```

Comment:
No need to split the string. `m_CurrentDirectory` is an std::filesystem::path which you can easily iterate over its elements with a range-for loop